cmd/wallet-js-sdk/src/oidc/presentation/openid4vp.js

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+Copyright Avast Software. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+import axios from "axios";
+import { decode } from "js-base64";
+import { CredentialManager, DIDManager } from "@";
+import * as jose from "jose";
+
+/**
+ *  oidc module is the oidc client that provides APIs for OIDC4VC flows.
+ *
+ *  @module oidc
+ *
+ */
+
+/**
+ * Client providing support for OIDC4VC methods.
+ *
+ * @alias module:oidc
+ */
+export class OpenID4VP {
+  // Creates a new OpenID4VP client
+  constructor({ agent, authToken, user }) {
+    if (!agent) {
+      throw new TypeError(
+        "Error initializing OIDC client: agent cannot be empty"
+      );
+    } else if (!authToken) {
+      throw new TypeError(
+        "Error initializing OIDC client: authToken cannot be empty"
+      );
+    } else if (!user) {
+      throw new TypeError(
+        "Error initializing OIDC client: user cannot be empty"
+      );
+    }
+    this.authToken = authToken;
```

Comment:
not a best practice to persist auth token as an attribute, should be passed dynamically while doing each function calls.